### DISCUSSION TITLE
作業0透過Clone至VScode產生錯誤

### DISCUSSION THREAD

**Discussion by wh20502** [1101-GS3073-C/Python-Community-GS3073-C#20] [Q&A]
當我按照老師的方法，將作業0的code網址複製，在終端機上面輸入 ：
git clone https://github.com/1101-GS3073-C/hw0-wh20502.git 
時，會跑出以下錯誤：

git : 無法辨識 'git' 詞彙是否為 Cmdlet、函數、指令檔或可執行程式的名稱。請檢查名稱拼字是否正
確，如果包含路徑的話，請確認路徑是否正確，然後再試一次。
位於 線路:1 字元:1
+ git clone https://github.com/1101-GS3073-C/hw0-wh20502.git
+ ~~~
    + CategoryInfo          : ObjectNotFound: (git:String) [], CommandNotFoundException       
    + FullyQualifiedErrorId : CommandNotFoundException

另外，因為我把VScode放在資料夾內，如果輸入： cd .\desktop\  。會跑出找不到該文件夾。
跑出來的錯誤訊息為：

PS C:\Users\User\Desktop\AI\VSCode> cd .\desktop\
cd : 找不到 'C:\Users\User\Desktop\AI\VSCode\desktop\' 路徑，因為它不存在。
位於 線路:1 字元:1
+ cd .\desktop\
+ ~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\User\Desktop\AI\VSCode\desktop\:Stri

請問有人有遇到一樣的問題，或是知道怎麼解決的嗎 ?

**Comment by Lynn12378** [comment]
不知道這個能不能幫到你><
我自己是windows 不知道mac要不要另外下載git
你可以先在終端機輸入git --version 檢查一下有沒有安裝git了
如果沒有下面網址有教怎麼安裝git
https://w3c.hexschool.com/git/fd6f6be

**Reply by wh20502** [reply]
我的作業軟體也是Win10。QQ

**Comment by handsomeniko** [comment]
你說輸入cd .\desktop\ 會有問題是因為你已經在"桌面(Desktop)"的資料夾裡面了，你看你的路徑是 'C:\Users\User\Desktop\AI\VSCode\，前面有出現Desktop對吧~ 其實你的作業要clone在哪個位置都沒有關係，老師放在桌面只是讓大家清楚知道檔案在哪裡而已，如果你也想放在桌面，你可以在終端機打cd ../.. ，回到桌面後再輸入git clone 網址，這樣你的作業就會出現在桌面了，如果你不確定你現在的位置在哪一層，可以輸入pwd，來看自己現在在哪裡，希望有幫助到你。

**Comment by chia-kai-chang** [comment]
@wh20502 
## 錯誤訊息：git : 無法辨識 'git' 詞彙是否為 Cmdlet、函數、指令檔或可執行程式的名稱
你的電腦還沒安裝git，如果是windows的電腦可以到這裡下載：https://git-scm.com/download/win

## 如果輸入： cd .\desktop\ 。會跑出找不到該文件夾
`cd .\desktop\` 會去查找你目前的 **工作目錄** 下，有沒有 `desktop` 資料夾，如果沒有，就會出現找不到資料夾的訊息。

根據你提供的資訊，要下這個指令：`cd C:\Users\User\Desktop` 才會切換到桌面的資料夾。

@handsomeniko 說的沒錯，老師只是為了示範，把git clone下來的資料夾放到桌面。實際上你們可以放在想要放置的路徑上。
